api/prisma/createTopics.ts

```diff
@@ -0,0 +1,286 @@
+    newParentCreationId1?: string;
+    newParentCreationId2?: string;
+    multipleParents: boolean;
+};
+
+const readDataFromFile = (): { data: CreateTopicData[]; error?: string } => {
+    const inputFileName = 'prisma/createTopics.json';
+
+    if (!fs.existsSync(inputFileName)) {
+        return {
+            data: [],
+            error: `Input file "${inputFileName}" not found.`
+        };
+    }
+
+    const fileData = JSON.parse(fs.readFileSync(inputFileName, 'utf8'));
+
+    return {
+        data: fileData
+    };
+};
+
+const validateData = async (
+    data: CreateTopicData[],
+    environment: 'int' | 'prod' = 'int',
+    dryRun?: boolean
+): Promise<{ valid: boolean; messages: string[] }> => {
+    let valid = true;
+    const messages: string[] = [];
+
+    for (const item of data) {
+        // Check that we have enough fields to link to parents
+        if (item.multipleParents) {
+            // One of the following combinations must be true to be valid
+            if (
+                !(
+                    // Both parents are also in this data
+                    (
+                        (item.newParentCreationId1 && item.newParentCreationId2) ||
+                        // Both parents already exist prior to running the script
+                        (item.intExistingParentId1 &&
+                            item.intExistingParentId2 &&
+                            item.prodExistingParentId1 &&
+                            item.prodExistingParentId2) ||
+                        // One parent already exists, one is also in this data
+                        (item.newParentCreationId1 && item.intExistingParentId1 && item.prodExistingParentId1)
+                    )
+                )
+            ) {
+                valid = false;
+                messages.push(
+                    `Creation ID ${item.creationId} should have multiple parents but insufficient IDs supplied`
+                );
+            }
+        } else {
+            if (!(item.newParentCreationId1 || (item.intExistingParentId1 && item.prodExistingParentId1))) {
+                valid = false;
+                messages.push(`Creation ID ${item.creationId} does not have parent ID(s) specified`);
+            }
+        }
+
+        // Check that creation ID of new parents exists in data
+        if (
+            item.newParentCreationId1 &&
+            !data.find((potentialParent) => potentialParent.creationId === item.newParentCreationId1)
+        ) {
+            valid = false;
+            messages.push(`Creation ID ${item.creationId} has an invalid new parent creation ID 1`);
+        }
+
+        if (
+            item.newParentCreationId2 &&
+            !data.find((potentialParent) => potentialParent.creationId === item.newParentCreationId2)
+        ) {
+            valid = false;
+            messages.push(`Creation ID ${item.creationId} has an invalid new parent creation ID 2`);
+        }
+
+        if (!dryRun) {
```

Comment:
Could do this even if not dry run? It isn't updating anything

Reply:
I'll try this again 😂 

Could potentially do this whether dry run or not. It's only finding things, it isn't updating anything

Reply:
Yeah, good point. I think working locally I was conflating "dry run" with "not touching the database" which isn't strictly true. I'll remove this check